silx/opencl/sift/alignment.py

```diff
@@ -168,11 +168,11 @@ def _free_buffers(self):
         """
         free all memory allocated on the device
         """
-        for buffer_name in self.cl_mem:
+        for buffer_name in list(self.cl_mem.keys()):
             if self.cl_mem[buffer_name] is not None:
                 try:
-                    del self.cl_mem[buffer_name]
-                    self.cl_mem[buffer_name] = None
+                    buffer = self.cl_mem.pop(buffer_name)
+                    del buffer
```

Comment:
Strangely the error `RuntimeError: dictionary keys changed during iteration` should have happened in every previous versions of Python, but didn't. 
This correction should indeed fix the problem.